common/src/main/java/uk/gov/justice/services/common/configuration/AbstractValueProducer.java

```diff
@@ -14,15 +15,19 @@ protected AbstractValueProducer() throws NamingException {
     }
 
     protected String jndiValueFor(final CommonValueAnnotationDef annotation) throws NamingException {
-        try {
-            return (String) initialContext.lookup(jndiNameFrom(annotation));
-        } catch (NameNotFoundException e) {
-            if (CommonValueAnnotationDef.NULL_DEFAULT.equals(annotation.defaultValue())) {
-                throw new MissingPropertyException(format("Missing property: %s", annotation.key()));
+        for (String jndiName : jndiNamesFrom(annotation)) {
+            try {
+                return (String) initialContext.lookup(jndiName);
```

Comment:
if we have 2 loops, won't the return exit out of the loop?
Is this purely going to work because the order of the array (app then global) ?